q2_sample_classifier/tests/test_classifier.py

```diff
@@ -84,31 +84,114 @@ def _load_md(md_fp):
             md = qiime2.Metadata(md)
             return md
 
+        def _load_mdc(md_fp, category):
+            md_fp = self.get_data_path(md_fp)
+            md = pd.DataFrame.from_csv(md_fp, sep='\t')
+            md = qiime2.MetadataCategory(md[category])
+            return md
+
         self.table_chard_fp = _load_df('chardonnay.table.qza')
         self.md_chard_fp = _load_md('chardonnay.map.txt')
+        self.mdc_chard_fp = _load_mdc('chardonnay.map.txt', 'Region')
         self.table_ecam_fp = _load_df('ecam-table-maturity.qza')
         self.md_ecam_fp = _load_md('ecam_map_maturity.txt')
+        self.mdc_ecam_fp = _load_mdc('ecam_map_maturity.txt', 'month')
 
+    # test that the plugin/visualizer work
     def test_classify_samples(self):
+        tmpd = join(self.temp_dir.name, 'RandomForestClassifier')
+        mkdir(tmpd)
+        classify_samples(tmpd, self.table_chard_fp, self.mdc_chard_fp,
+                         test_size=0.5, cv=3,
+                         n_estimators=2, n_jobs=-1,
+                         estimator='RandomForestClassifier',
+                         parameter_tuning=True,
+                         optimize_feature_selection=True)
+
+    # test that each classifier works and delivers an expected accuracy result
+    # when a random seed is set.
+    def test_classifiers(self):
         for classifier in ['RandomForestClassifier', 'ExtraTreesClassifier',
                            'GradientBoostingClassifier', 'AdaBoostClassifier',
-                           'KNeighborsClassifier', 'LinearSVC', 'SVC']:
+                           'LinearSVC', 'SVC', 'KNeighborsClassifier']:
             tmpd = join(self.temp_dir.name, classifier)
             mkdir(tmpd)
-            classify_samples(tmpd, self.table_chard_fp, self.md_chard_fp,
-                             category='Vineyard', test_size=0.5, cv=3,
-                             n_estimators=2, n_jobs=-1, estimator=classifier)
-
+            estimator, pd, pt = _set_parameters_and_estimator(
+                classifier, self.table_chard_fp, self.md_chard_fp, 'Region',
+                n_estimators=10, n_jobs=-1, cv=1,
```

Comment:
Can you do `njobs=1` throughout these tests? I'm worried about the test taking over the resources on someone's machine (e.g., if someone were to run this on the login node of their cluster, we wouldn't want it to use all CPUs).